### PR TITLE
Remove envoy_reloadable_features_initialize_upstream_filters

### PR DESCRIPTION
**Pull request by sayboras**
Commit Message: Remove envoy_reloadable_features_initialize_upstream_filters
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] #30421
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by sayboras** [comment]
cc @alyssawilk whenever you have time, thanks a lot.

File: changelogs/current.yaml [hunk 1]
```diff
@@ -214,3 +214,6 @@ deprecated:
   change: |
     deprecated runtime key ``overload.global_downstream_max_connections`` in favor of :ref:`downstream connections monitor
     <envoy_v3_api_msg_extensions.resource_monitors.downstream_connections.v3.DownstreamConnectionsConfig>`.
+- area: upstream
```
Comment (adisuissa) [inline changelogs/current.yaml:217]: Please move to hte `removed_config_or_runtime` section above.

File: test/integration/cluster_filter_integration_test.cc [hunk 2]
```diff
@@ -96,28 +96,11 @@ class PoliteFilterConfigFactory
...
-  return fmt::format(
-      "{}_{}",
-      TestUtility::ipTestParamsToString(
-          testing::TestParamInfo<Network::Address::IpVersion>(std::get<0>(params.param), 0)),
-      std::get<1>(params.param) ? "do_initialize_upstream_filters"
-                                : "dont_initialize_upstream_filters");
-}
-
 class ClusterFilterIntegrationTestBase
     : public testing::TestWithParam<std::tuple<Network::Address::IpVersion, bool>>,
```
Comment (adisuissa) [inline test/integration/cluster_filter_integration_test.cc:100]: I think the 2nd param (`bool`) can be removed, no?

**Review by adisuissa — COMMENTED** [review]
Thanks for cleaning this up!
I've left a couple of minor comments, but otherwise LGTM!

File: changelogs/current.yaml (hunk 1 above)
Comment (sayboras) [reply changelogs/current.yaml:217]: ah right, good spot.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

/assign-from @envoyproxy/senior-maintainers

**Pull request merged**